src/xcode/ENA/ENA/Source/RootCoordinator.swift

```diff
@@ -245,3 +246,51 @@ extension RootCoordinator: ENStateHandlerUpdating {
 		}
 	}
 }
+
+// MARK: - Protocol UITabBarControllerDelegate
+
+extension RootCoordinator: UITabBarControllerDelegate {
+	func tabBarController(_ tabBarController: UITabBarController, shouldSelect viewController: UIViewController) -> Bool {
+		if viewController == tabBarController.selectedViewController {
+			if let naviVC = viewController as? UINavigationController {
+				naviVC.scrollEmbeddedTableToTop()
+			}
+		}
+		return true
+	}
+}
+
+extension UINavigationController {
+	// only if on top level of the navigation hierarchy: scroll the embedded table view up
+	func scrollEmbeddedTableToTop() {
+		guard !isEditing && presentedViewController == nil && viewControllers.count == 1 else {
+			return
+		}
+		if let tableView = topViewController?.embeddedTableView {
+			scrollTableToTop(tableView)
+		}
+	}
+
+	private func scrollTableToTop(_ tableView: UITableView) {
+		guard !tableView.isEditing else {
+			return
+		}
+		tableView.scrollToRow(at: IndexPath(indexes: [0, 0]), at: .top, animated: true)
+	}
+}
+
+private extension UIViewController {
+	var embeddedTableView: UITableView? {
+		([view] + view.subviews).first(ofType: UITableView.self)
+	}
+}
+
+private extension Array {
+	func first<T>(ofType _: T.Type) -> T? {
+		first(where: { $0 is T }) as? T
+	}
+
+	func last<T>(ofType _: T.Type) -> T? {
+		last(where: { $0 is T }) as? T
+	}
```

Comment:
I can't see that `last` gets used - if not we should remove it.

Reply:
@KaiTeuber 
Indeed.
Another thought. Actually, I copy-pasted this from AppNavigationController.
Maybe I should move the new code over to AppNavigationController? Then I don't need to duplicate it?
In that case AppNavigationController would take responsibility as delegate for the tab bar?

Reply:
@KaiTeuber I moved now the solution into AppNavigationController:
- AppNavigationController takes now responsibility as delegate for the tab bar
- Array extensions already available here, no need to duplicate them